plugins/dap/dap-ui.nix

```diff
@@ -0,0 +1,209 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.dap.extensions.dap-ui;
+  helpers = import ../helpers.nix {inherit lib;};
+
+  mkSizeOption =
+    helpers.mkNullOrOption
+    (
+      with types;
+        either int (numbers.between 0.0 1.0)
+    );
+
+  mkMappingOption =
+    helpers.defaultNullOpts.mkNullable
+    (
+      with types;
+        attrsOf (either str (listOf str))
+    );
+
+  layoutOption = types.submodule {
+    options = {
+      elements = mkOption {
+        default = [];
+        description = "Elements to display in this layout.";
+        type = types.listOf types.attrs;
```

Comment:
This is able to be simply a string list ([source](https://github.com/rcarriga/nvim-dap-ui/blob/c020f660b02772f9f3d11f599fefad3268628a9e/lua/dapui/config/init.lua#L29))
Instead of an attr, maybe a submodule with `id` & `size`